### PR TITLE
Feature/myst 708 kill switch

### PR DESCRIPTION
**Pull request by zolia**
tried to find middle ground between @Waldz and @tadovas remarks.

### CODE REVIEW

File: core/connection/interface.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ package connection
 
 import (
 	"github.com/mysteriumnetwork/node/communication"
+	dto2 "github.com/mysteriumnetwork/node/core/node/dto"
```
Comment (tadovas) [inline core/connection/interface.go]: avoid generic packages like DTO

File: core/connection/interface.go [hunk 2]
```diff
@@ -31,12 +32,12 @@ type DialogCreator func(consumerID, providerID identity.Identity, contact dto.Co
 
 // VpnClientCreator creates new vpn client by given session,
 // consumer identity, provider identity and uses state callback to report state changes
-type VpnClientCreator func(session.SessionDto, identity.Identity, identity.Identity, state.Callback) (openvpn.Process, error)
+type VpnClientCreator func(session.SessionDto, identity.Identity, identity.Identity, state.Callback, dto2.ConnectOptions) (openvpn.Process, error)
```
Comment (tadovas) [inline core/connection/interface.go]: Connection manager should not depend on external Connect options. At least define them in the same package

File: openvpn/client_config.go [hunk 3]
```diff
@@ -26,6 +27,13 @@ type ClientConfig struct {
 	*config.GenericConfig
 }
 
+// SetConnectOptions sets connect options provided through endpoint parameters
+func (c *ClientConfig) SetConnectOptions(options dto.ConnectOptions) {
```
Comment (tadovas) [inline openvpn/client_config.go]: Client config should have clear methods -> RestrictReconnects and so on, dont add external dependency as ConnectOptions

File: core/connection/interface.go (hunk 1 above)
Comment (Waldz) [reply core/connection/interface.go]: I'd like `dto69` more :)

File: core/connection/manager.go [hunk 4]
```diff
@@ -71,7 +72,7 @@ func NewManager(mysteriumClient server.Client, dialogCreator DialogCreator,
 	}
 }
 
-func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) (err error) {
+func (manager *connectionManager) Connect(consumerID, providerID identity.Identity, options dto2.ConnectOptions) (err error) {
```
Comment (Waldz) [inline core/connection/manager.go]: +100 for DTO, I am doing same wiith `serviceOptions` for vpn server

File: core/node/dto/connect_options.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package dto
+
+// ConnectOptions holds tequilapi connect options
+// swagger:model ConnectOptionsDTO
+type ConnectOptions struct {
```
Comment (Waldz) [inline core/node/dto/connect_options.go]: Lets have this DTO in `connection` package

File: e2e/wallet_helper.go [hunk 6]
```diff
@@ -35,6 +32,9 @@ import (
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/params"
 	"github.com/mysteriumnetwork/node/tequilapi/client"
+	"github.com/mysteriumnetwork/payments/cli/helpers"
+	mysttoken "github.com/mysteriumnetwork/payments/mysttoken/generated"
```
Comment (Waldz) [inline e2e/wallet_helper.go]: +1 good idea

File: openvpn/client_config.go [hunk 7]
```diff
@@ -73,14 +79,15 @@ func defaultClientConfig(runtimeDir string, scriptSearchPath string) *ClientConf
...
-func NewClientConfigFromSession(vpnConfig *VPNConfig, configDir string, runtimeDir string) (*ClientConfig, error) {
+func NewClientConfigFromSession(vpnConfig *VPNConfig, configDir string, runtimeDir string, options dto.ConnectOptions) (*ClientConfig, error) {
 
 	err := NewDefaultValidator().IsValid(vpnConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	clientFileConfig := newClientConfig(runtimeDir, configDir)
+	clientFileConfig.SetConnectOptions(options)
```
Comment (Waldz) [inline openvpn/client_config.go]: You should map `ConnectionOptions` -> `openvpn.config` here

File: tequilapi/client/client.go [hunk 8]
```diff
@@ -100,13 +102,17 @@ func (client *Client) RegistrationStatus(address string) (RegistrationStatusDTO,
 }
 
 // Connect initiates a new connection to a host identified by providerID
-func (client *Client) Connect(consumerID, providerID string) (status StatusDTO, err error) {
+func (client *Client) Connect(consumerID, providerID string, disableKill bool) (status StatusDTO, err error) {
```
Comment (Waldz) [inline tequilapi/client/client.go]: Lets have pass all `dto.ConnectOptions` as argument

File: tequilapi/endpoints/connection.go [hunk 9]
```diff
@@ -46,6 +47,10 @@ type connectionRequest struct {
 	// required: true
 	// example: 0x0000000000000000000000000000000000000002
 	ProviderID string `json:"providerId"`
+
+	// connect options
+	// required: false
+	ConnectOptions dto.ConnectOptions `json:"connectOptions,omitempty"`
```
Comment (Waldz) [inline tequilapi/endpoints/connection.go]: It's dont have to repeat `connectOptions` in `connectionRequest`. IMHO `options` would be enought in REST contract

File: core/connection/interface.go [hunk 10]
```diff
@@ -31,12 +32,12 @@ type DialogCreator func(consumerID, providerID identity.Identity, contact dto.Co
...
 
 // Manager interface provides methods to manage connection
 type Manager interface {
 	// Connect creates new connection from given consumer to provider, reports error if connection already exists
-	Connect(consumerID identity.Identity, providerID identity.Identity) error
+	Connect(consumerID identity.Identity, providerID identity.Identity, options dto2.ConnectOptions) error
```
Comment (Waldz) [inline core/connection/interface.go]: +100 for separating openvopn specific things to separate struct

File: tequilapi/client/client.go [hunk 11]
```diff
@@ -100,13 +102,17 @@ func (client *Client) RegistrationStatus(address string) (RegistrationStatusDTO,
...
+	connectOptions := dto.ConnectOptions{DisableKillSwitch: disableKill}
+
 	payload := struct {
-		Identity   string `json:"consumerId"`
-		ProviderID string `json:"providerId"`
+		Identity       string             `json:"consumerId"`
+		ProviderID     string             `json:"providerId"`
+		ConnectOptions dto.ConnectOptions `json:"connectOptions"`
```
Comment (Waldz) [inline tequilapi/client/client.go]: It's dont have to repeat `connectOptions` in `connectionRequest`. IMHO `options` would be enought in REST contract

**Comment by zolia** [comment]
Branch rebased on master

**Pull request merged**